### PR TITLE
feat: supbase auth=> updated users.py to use supabase auth

### PR DESCRIPTION
**Pull request by sway4em**
**Trello card**: https://trello.com/c/IU1gl35j

- Integrated Supabase OAuth into the users endpoint
- Setup reusable auth middleware
- Updated the balance endpoints to use authentication

### CODE REVIEW

**Comment by sway4em** [comment]
[Integrate User Endpoint with Supabase OAuth](https://trello.com/c/IU1gl35j/16-integrate-user-endpoint-with-supabase-oauth)

File: src/api/users.py [hunk 1]
```diff
@@ -1,81 +1,225 @@
...
 router = APIRouter(
     prefix="/users",
     tags=["users"],
 )
 
+security = HTTPBearer()
 
-class User(BaseModel):
+# model definitions
+# DO NOT MOVE THESE, MUST BE DEFINED BEFORE THE ROUTES
```
Comment (Axelph237) [inline src/api/users.py:26]: I am going to move this

File: src/api/users.py [hunk 2]
```diff
@@ -1,81 +1,225 @@
...
+        raise HTTPException(
+            status_code=401,
+            detail="Invalid authentication credentials",
+            headers={"WWW-Authenticate": "Bearer"},
+        )
 
+@router.post("/register")
+async def register(user: UserRegister):
+    try:
+        # register with supabase auth
```
Comment (Axelph237) [inline src/api/users.py:57]: Is this synchronous?

**Review by Axelph237 — APPROVED** [review]
Looks good!

**Pull request merged**